cypher-shell/src/main/java/org/neo4j/shell/commands/Param.java

```diff
@@ -23,17 +23,26 @@
     private static final Pattern lambdaPattern = Pattern.compile("^\\s*(?<key>[\\p{L}_][\\p{L}0-9_]*)\\s*=>\\s*(?<value>.+)$");
     private static final Pattern lambdaMapPattern = Pattern.compile("^\\s*(?<key>[\\p{L}_][\\p{L}0-9_]*):\\s*=>\\s*(?<value>.+)$");
 
-    public static final String COMMAND_NAME = ":param";
+    private final String commandName;
+    private final boolean fromWithinShell;
     private final ParameterMap parameterMap;
 
-    public Param(@Nonnull final ParameterMap parameterMap) {
+    /**
+     * @param parameterMap the map to set parameters in
+     * @param fromWithinShell If {@code true}, this param evaluates ":param" commands from within the shell.
+     *                        If {@code false}, this param evaluates "--param" commands from CLI arguments.
+     */
+    public Param(@Nonnull final ParameterMap parameterMap,
+                 final boolean fromWithinShell) {
```

Comment:
It feels a bit hacky to jack the command line parameter into the command with a boolean like this. It would be nicer to have a base class for the shared functionality of parsing and setting a parameter map (between 
ArgumentAction and Command), and leave the `Command` class as it is. I don't think `getName` and `getUsage` has any usages for command line argument actions.